src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,537 @@
+        /// </summary>
+        /// <param name="left">The column position from left to right starting with 0.</param>
+        /// <param name="top">The row position from the top starting with 0.</param>
+        /// <exception cref="ArgumentOutOfRangeException">The new position negative.</exception>
+        public void SetCursorPosition(int left, int top)
+        {
+            lock (_lock)
+            {
+                if (left < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(left));
+                }
+                if (top < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(top));
+                }
+                if (left < Size.Width && top < Size.Height)
+                {
+                    _lcd.SetCursorPosition(left, top);
+                }
+                CursorLeft = left;
+                CursorTop = top;
+            }
+        }
+
+        /// <summary>
+        /// Write text to display.
+        /// </summary>
+        /// <remarks>
+        /// There are only 256 characters available. There are chip variants
+        /// with different character sets. Characters from space ' ' (32) to
+        /// '}' are usually the same with the exception of '\', which is a
+        /// yen symbol on some chips '¥'.
+        /// </remarks>
+        /// <param name="text">Text to be displayed.</param>
+        public void Write(string text)
+        {
+            text = text.Replace("\r\n", "\n"); // Change to linux format only, so we have to consider only this case further
+
+            List<string> lines = text.Split("\n", StringSplitOptions.None).ToList();
+            FindLineWraps(CursorLeft, lines);
+            for (int i = 0; i < lines.Count; i++)
+            {
+                string line = lines[i];
+                string currentLine = line;
+
+                int remainingChars = Math.Min(currentLine.Length, Size.Width - CursorLeft);
+                if (remainingChars > 0)
+                {
+                    lock (_lock)
+                    {
+                        WriteCurrentLine(currentLine.Substring(0, remainingChars));
+                    }
+                }
+
+                if (i != lines.Count - 1)
+                {
+                    // Insert newline except after the last text segment.
+                    NewLine();
+                }
+            }
+
+        }
+
+        /// <summary>
+        /// Asynchronously writes text to the display.
+        /// See <see cref="Write(string)"/> for limitations.
+        /// Only one write operation (synchronous or asynchronous) should be executed at once, otherwise the execution order 
+        /// is undefined. 
+        /// </summary>
+        /// <param name="text">Text to write</param>
+        /// <returns>A task object</returns>
+        public Task WriteAsync(string text)
```

Comment:
don't add Async version if all it's doing is starting new task

Reply:
I don't understand. The Write operation can take very long (especially if the NewLineSleepTime is big), so why would an async version not be useful?

Reply:
@pgrawehr the only time you should be doing Async version if you actually are trully async (i.e. have a dedicated thread always running in the background and picking up jobs and later marking task as completed) or you are trying to match existing interface. In here user can call Task.Run themselves so I believe we should not provide such overload.